modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/appcatalog/UserResourceProfileRepository.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    protected String saveUserResourceProfileData(UserResourceProfile userResourceProfile) throws AppCatalogException {
+        UserResourceProfileEntity userResourceProfileEntity = saveUserResourceProfile(userResourceProfile);
+        return userResourceProfileEntity.getUserId();
+    }
+
+    protected UserResourceProfileEntity saveUserResourceProfile(UserResourceProfile userResourceProfile) throws AppCatalogException {
+        String userId = userResourceProfile.getUserId();
+        String gatewayId = userResourceProfile.getGatewayID();
+        Mapper mapper = ObjectMapperSingleton.getInstance();
+        UserResourceProfileEntity userResourceProfileEntity = mapper.map(userResourceProfile, UserResourceProfileEntity.class);
+
+        if (userResourceProfileEntity.getUserComputeResourcePreferences() != null) {
+            logger.debug("Populating the Primary Key UserComputeResourcePreferences objects for the User Resource Profile");
+            userResourceProfileEntity.getUserComputeResourcePreferences().forEach(userComputeResourcePreferenceEntity -> userComputeResourcePreferenceEntity.setUserId(userId));
+            userResourceProfileEntity.getUserComputeResourcePreferences().forEach(userComputeResourcePreferenceEntity -> userComputeResourcePreferenceEntity.setGatewayId(gatewayId));
+        }
+
+        if (userResourceProfileEntity.getUserStoragePreferences() != null) {
+            logger.debug("Populating the Primary Key UserStoragePreferences objects for the User Resource Profile");
+            userResourceProfileEntity.getUserStoragePreferences().forEach(userStoragePreferenceEntity -> userStoragePreferenceEntity.setUserId(userId));
+            userResourceProfileEntity.getUserStoragePreferences().forEach(userStoragePreferenceEntity -> userStoragePreferenceEntity.setGatewayId(gatewayId));
+        }
+
+        if (!isUserResourceProfileExists(userId, gatewayId)) {
+            logger.debug("Checking if the User Resource Profile already exists");
+            userResourceProfileEntity.setCreationTime(new Timestamp(System.currentTimeMillis()));
+        }
+
+        userResourceProfileEntity.setUpdateTime(new Timestamp(System.currentTimeMillis()));
+        return execute(entityManager -> entityManager.merge(userResourceProfileEntity));
+    }
+
+    @Override
+    public String addUserResourceProfile(UserResourceProfile userResourceProfile) throws AppCatalogException {
+        return saveUserResourceProfileData(userResourceProfile);
+    }
+
+    @Override
+    public void updateUserResourceProfile(String userId, String gatewayId, UserResourceProfile updatedProfile) throws AppCatalogException {
+        saveUserResourceProfileData(updatedProfile);
+    }
+
+    @Override
+    public UserResourceProfile getUserResourceProfile(String userId, String gatewayId) throws AppCatalogException {
+        Map<String, Object> queryParameters = new HashMap<>();
```

Comment:
Why can't this create a UserResourceProfilePK and then issue a get() with that id?  This is just looking up by the primary key.